revisions/api/handlers/list.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/web-platform-tests/wpt.fyi/revisions/announcer"
+	"github.com/web-platform-tests/wpt.fyi/revisions/api"
+	"github.com/web-platform-tests/wpt.fyi/revisions/epoch"
+)
+
+// ListHandler handles HTTP requests for listing epochal revisions.
+func ListHandler(a api.API, w http.ResponseWriter, r *http.Request) {
+	ancr := a.GetAnnouncer()
+	if ancr == nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		w.Write(a.ErrorJSON("Announcer not yet initialized"))
+		return
+	}
+
+	epochs := a.GetEpochs()
+	if len(epochs) == 0 {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(a.ErrorJSON("No epochs"))
+		return
+	}
+
+	q := r.URL.Query()
+
+	numRevisions := 1
+	if nr, ok := q["num_revisions"]; ok {
```

Comment:
- `Query` has a `Get("num_revisions")` method which returns an empty string if missing.
- Probably just parse as an int, then assert size limits.
- Move the `num_revisions` param parsing to the `params.go` file

Reply:
I don't quite follow on this one. Let's sync offline.

Reply:
Ignore the last 2 items there - I misread the `len(1)` assertion as being the number of chars lol.
But yeah, no need to fail for multiple copies - `Get` will only give you the last one.